ThunkLibs/include/common/Guest.h

```diff
@@ -121,6 +122,14 @@ inline Result CallHostFunction(Args... args) {
   // This magic incantation also works in that instance so this is about the best we can do without adding a new attribute to clang for modifying the
   // ABI.
   asm volatile("" : "=r" (host_addr));
+#else
+  // mm0 register is chosen as the register that contains the host_addr.
+  // This register is chosen as it doesn't conflict with vectorcall.
+  // This register also overlaps the x87 st(0) register which is used as float value return.
+  // st(0) is required to be empty coming in to the function because of this.
+  register uintptr_t host_addr asm ("mm0");
```

Comment:
Has this been tested, or should I hack up some unit tests to verify this produces working assembly?

Reply:
This has been tested to get through at least one function call correctly, and inspected the output assembly to ensure something wasn't breaking it.
glXGetVisual or something was working.